spec.html

```diff
@@ -36280,25 +36739,86 @@ <h1>
           <emu-alg>
             1. Let _basicWordChars_ be the CharSet containing every character in the ASCII word characters.
             1. Let _extraWordChars_ be the CharSet containing all characters _c_ such that _c_ is not in _basicWordChars_ but Canonicalize(_rer_, _c_) is in _basicWordChars_.
-            1. Assert: _extraWordChars_ is empty unless _rer_.[[Unicode]] is *true* and _rer_.[[IgnoreCase]] is *true*.
+            1. Assert: _extraWordChars_ is empty unless _rer_.[[EitherUnicode]] is *true* and _rer_.[[IgnoreCase]] is *true*.
             1. Return the union of _basicWordChars_ and _extraWordChars_.
           </emu-alg>
         </emu-clause>
 
+        <emu-clause id="sec-allcharacters" type="abstract operation">
+          <h1>
+            AllCharacters (
+              _rer_: a RegExp Record,
+            ): a CharSet
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>Returns the set of “all characters” according to the regular expression flags.</dd>
+          </dl>
+          <emu-alg>
+            1. If _rer_.[[UnicodeSets]] is *true* and _rer_.[[IgnoreCase]] is *true*, then
+              1. Return the set of all Unicode code points _c_ that do not have a <a href="https://www.unicode.org/reports/tr44/#Simple_Case_Folding">Simple Case Folding</a> mapping (that is, scf(_c_)=_c_).
+            1. Else if _rer_.[[EitherUnicode]] is *true*, then
+              1. Return the set containing all code point values.
+            1. Else,
+              1. Return the set containing all code unit values.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-maybesimplecasefolding" type="abstract operation">
+          <h1>
+            MaybeSimpleCaseFolding (
+              _rer_: a RegExp Record,
+              _A_: a CharSet,
+            ): a CharSet
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>If _rer_.[[UnicodeSets]] is *false* or _rer_.[[IgnoreCase]] is *false*, it returns _A_. Otherwise, it uses the <a href="https://www.unicode.org/reports/tr44/#Simple_Case_Folding">Simple Case Folding</a> (<dfn>scf</dfn>) definitions in the file <a href="https://unicode.org/Public/UCD/latest/ucd/CaseFolding.txt"><code>CaseFolding.txt</code></a> of the Unicode Character Database (each of which maps a single code point to another single code point) to map each CharSetElement of _A_ character-by-character into a canonical form and returns the resulting CharSet.</dd>
+          </dl>
+          <emu-alg>
+            1. If _rer_.[[UnicodeSets]] is *false* or _rer_.[[IgnoreCase]] is *false*, return _A_.
+            1. Let _B_ be an empty CharSet.
+            1. For each CharSetElement _s_ of _A_, do
+              1. Let _t_ be an empty sequence of characters.
+              1. For each single code point _cp_ in _s_, do
+                1. Append scf(_cp_) to _t_.
```

Comment:
Similar `ecmarkup` issue here:

```
error: could not find definition for scf (typecheck) at spec.html:36784:27:
  36782 |               1. Let _t_ be an empty sequence of characters.
  36783 |               1. For each single code point _cp_ in _s_, do
> 36784 |                 1. Append scf(_cp_) to _t_.
        |                           ^
  36785 |               1. Set _B_ to the union of _B_ and _t_.
  36786 |             1. Return _B_.
  36787 |           </emu-alg>
```

@tc39/ecma262-editors any suggestions on how to properly define `scf`?

Reply:
This has been addressed.